mitosheet/mitosheet/public/v3/sheet_functions/bool_functions.py

```diff
@@ -60,10 +59,73 @@ def AND(*argv: BoolInputType) -> BoolFunctionReturnType:
         lambda previous_series, new_series: previous_series & new_series
     )
 
+@cast_values_in_arg_to_type('series', 'bool')
+@handle_sheet_function_errors
+def BOOL(series: BoolRestrictedInputType) -> BoolFunctionReturnType:
+    """
+    {
+        "function": "BOOL",
+        "description": "Converts the passed arguments to boolean values, either True or False. For numberic values, 0 converts to False while all other values convert to True.",
+        "search_terms": ["bool", "boolean", "true", "false", "dtype", "convert"],
+        "examples": [
+            "BOOL(Amount_Payed)",
+            "AND(BOOL(Amount_Payed), Is_Paying)"
+        ],
+        "syntax": "BOOL(series)",
+        "syntax_elements": [{
+                "element": "series",
+                "description": "An series to convert to boolean values, either True or False."
+            }
+        ]
+    }
+    """
+    if isinstance(series, bool):
+        return series
+
+    return series.fillna(False)
+
+@cast_values_in_arg_to_type('condition', 'bool')
+@handle_sheet_function_errors
+def IF(condition: pd.Series, true_series: AnyPrimitiveOrSeriesInputType, false_series: AnyPrimitiveOrSeriesInputType) -> pd.Series:
```

Comment:
Why does the `condition` have to be a series. If I set the `condition` as `3<4` then it would just be a primitive and cast_values_in_arg_to_type would return a primitive.


Similarly, the return type should be able to be a primitive value instead of a series.

Reply:
Static conditions are kinda nonsense for this function (you know the answer), so I think we can leave this as-is. In which case, it also returns a series.